src/compile/selection/selection.ts

```diff
@@ -202,6 +212,32 @@ export function assembleLayerSelectionMarks(model: LayerModel, marks: any[]): an
   return clipGroup ? clipMarks(marks) : marks;
 }
 
+const IDENTIFIER_TRANSFORM: VgIdentifierTransform = {type: 'identifier', as: SELECTION_ID};
+export function addIdentifierTransforms(model: Model, data: VgData[]) {
+  let identifier = false;
+  forEachSelection(model, (selCmpt) => {
+    identifier = identifier || selCmpt.project.some((proj) => proj.field === SELECTION_ID);
+  });
+
+  if (!identifier) {
+    return data;
+  }
+
+  data.forEach((def) => {
+    let index = -1;
+    const transform = def.transform || [];
+    const aggr = transform.some((t, i) => (index = i, t.type === 'aggregate'));
```

Comment:
I'm emulating [`array.findIndex`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) which we don't seem to have typings for unfortunately. We need to add an `identifier` transform to the `source` dataset and any time new tuples are derived (which, with Vega-Lite, I think only happens after an `aggregate`).